qtpy/tests/test_qtgui.py

```diff
@@ -177,6 +177,22 @@ def test_qtextcursor_moveposition():
     assert cursor.selectedText() == "foo bar baz"
 
 
+@pytest.mark.skipif(
+    sys.platform == "darwin" and sys.version_info[:2] == (3, 7),
+    reason="Stalls on macOS CI with Python 3.7",
+)
+def test_QAction_functions(qtbot):
+    """Test `QtGui.QAction.setShortcut` compatibility with Qt6 types."""
+    action = QtGui.QAction("QtPy", None)
+    action.setShortcut(QtGui.QKeySequence.UnknownKey)
+    action.setShortcuts([QtGui.QKeySequence.UnknownKey])
+    action.setShortcuts(QtGui.QKeySequence.UnknownKey)
+    action.setShortcut(QtCore.Qt.Key_F1)
+    action.setShortcuts([QtCore.Qt.Key_F1])
+    # The following line is wrong even for Qt6 == 6.6:
```

Comment:
Perhaps, it will. I see the Qt methods become more and more robust.

Moreover, the comments mark that the lines are skipped intentionally, not mistakenly. Should I rephrase the comment?

Reply:
Perhaps we should move this to another test, and mark it with [`@pytest.mark.xfail(strict=True)`](https://docs.pytest.org/en/7.1.x/how-to/skipping.html#strict-parameter) so we know when it starts working?

Reply:
Well, it's hard to make the fix short and fail here. The way I've fixed the functions makes the `action.setShortcuts(QtCore.Qt.Key_F1)` line work fine. Either I fail there intentionally (add an extra condition) or I leave it without a test for Qt < 6.5.